src/debug/dart_debug_impl.ts

```diff
@@ -221,14 +264,42 @@ export class DartDebugSession extends DebugSession {
 					this.sendEvent(new TerminatedEvent());
 			}, 100);
 		});
+
+		this.observatory.onError((error) => {
+			if (completer.promise) {
+				completer.resolve(error);
+				completer.promise = null;
+			}
+		});
+		return completer.promise;
 	}
 
 	protected disconnectRequest(
 		response: DebugProtocol.DisconnectResponse,
 		args: DebugProtocol.DisconnectArguments,
 	): void {
-		if (this.childProcess != null)
+		if (this.childProcess != null) {
 			this.childProcess.kill();
+		} else {
+			// Remove all breakpoints from the VM.
+			const removeBreakpointPromises = [];
+			for (const thread of this.threadManager.threads) {
+				removeBreakpointPromises.push(thread.removeAllBreakpoints());
+			}
+			Promise.all(removeBreakpointPromises).then((_) => {
+				// Restart any paused threads.
+				const resumePromises = [];
+				for (const thread of this.threadManager.threads) {
+					if (thread.paused) {
+						resumePromises.push(this.observatory.resume(thread.ref.id));
```

Comment:
This doesn't seem to be working for me; when I disconnect it remains paused.

Reply:
Hmm, can't repro now :)

Reply:
It could be that the thread.paused logic is incorrect in some cases. It worked in simple cases, but I didn't test it extensively.